File: client/scalaz/src/main/scala/org/openapitools/client/api/Coding.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import Coding._

case class Coding (
  /* Unique study key */
  studyKey: Option[String],
/* Name of the site associated with the coded data */
  siteName: Option[String],
/* Site ID associated with the coded data */
  siteId: Option[Integer],
/* Subject ID associated with the coded data */
  subjectId: Option[Integer],
/* Subject key (display ID) associated with the coded data */
  subjectKey: Option[String],
/* Form ID where the coded data originates */
  formId: Option[Integer],
/* Name of the form where the coded data originates */
  formName: Option[String],
/* Form key where the coded data originates */
  formKey: Option[String],
/* Revision number of the coding entry */
  revision: Option[Integer],
/* Record ID associated with the coded data */
  recordId: Option[Integer],
/* Variable name (field) that was coded */
  variable: Option[String],
/* Original value entered that required coding */
  value: Option[String],
/* Mednet coding ID */
  codingId: Option[Integer],
/* Standardized code assigned (e.g., dictionary term) */
  code: Option[String],
/* Name of the user who performed the coding */
  codedBy: Option[String],
/* Reason for coding or any notes on changes */
  reason: Option[String],
/* Name of the dictionary used (e.g., MedDRA) */
  dictionaryName: Option[String],
/* Version of the dictionary used */
  dictionaryVersion: Option[String],
/* Date when the coding was performed */
  dateCoded: Option[String])

object Coding {
  import DateTimeCodecs._

  implicit val CodingCodecJson: CodecJson[Coding] = CodecJson.derive[Coding]
  implicit val CodingDecoder: EntityDecoder[Coding] = jsonOf[Coding]
  implicit val CodingEncoder: EntityEncoder[Coding] = jsonEncoderOf[Coding]
}

File: API/Client/Coding.js
goog.provide('API.Client.Coding');

/**
 * @record
 */
API.Client.Coding = function() {}

/**
 * Unique study key
 * @type {!string}
 * @export
 */
API.Client.Coding.prototype.studyKey;

/**
 * Name of the site associated with the coded data
 * @type {!string}
 * @export
 */
API.Client.Coding.prototype.siteName;

/**
 * Site ID associated with the coded data
 * @type {!number}
 * @export
 */
API.Client.Coding.prototype.siteId;

/**
 * Subject ID associated with the coded data
 * @type {!number}
 * @export
 */
API.Client.Coding.prototype.subjectId;

/**
 * Subject key (display ID) associated with the coded data
 * @type {!string}
 * @export
 */
API.Client.Coding.prototype.subjectKey;

/**
 * Form ID where the coded data originates
 * @type {!number}
 * @export
 */
API.Client.Coding.prototype.formId;

/**
 * Name of the form where the coded data originates
 * @type {!string}
 * @export
 */
API.Client.Coding.prototype.formName;

/**
 * Form key where the coded data originates
 * @type {!string}
 * @export
 */
API.Client.Coding.prototype.formKey;

/**
 * Revision number of the coding entry
 * @type {!number}
 * @export
 */
API.Client.Coding.prototype.revision;

/**
 * Record ID associated with the coded data
 * @type {!number}
 * @export
 */
API.Client.Coding.prototype.recordId;

/**
 * Variable name (field) that was coded
 * @type {!string}
 * @export
 */
API.Client.Coding.prototype.variable;

/**
 * Original value entered that required coding
 * @type {!string}
 * @export
 */
API.Client.Coding.prototype.value;

/**
 * Mednet coding ID
 * @type {!number}
 * @export
 */
API.Client.Coding.prototype.codingId;

/**
 * Standardized code assigned (e.g., dictionary term)
 * @type {!string}
 * @export
 */
API.Client.Coding.prototype.code;

/**
 * Name of the user who performed the coding
 * @type {!string}
 * @export
 */
API.Client.Coding.prototype.codedBy;

/**
 * Reason for coding or any notes on changes
 * @type {!string}
 * @export
 */
API.Client.Coding.prototype.reason;

/**
 * Name of the dictionary used (e.g., MedDRA)
 * @type {!string}
 * @export
 */
API.Client.Coding.prototype.dictionaryName;

/**
 * Version of the dictionary used
 * @type {!string}
 * @export
 */
API.Client.Coding.prototype.dictionaryVersion;

/**
 * Date when the coding was performed
 * @type {!string}
 * @export
 */
API.Client.Coding.prototype.dateCoded;

